201901012 updatProject/Presenter/UpdateClientController.cs

```diff
@@ -0,0 +1,263 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using UpdateUpload.Interface;
+using UpdateUpload.Model;
+using WinSCP;
+
+namespace UpdateUpload.Presenter
+{
+    class UpdateClientController : IUpdateClientController
+    {
+        #region Variable
+
+        #region Constant
+
+        private const string VersionTxtFileName = "Version.txt";
+        private const string RequestUriString = "ftp://192.168.53.26/";
+        private const string UserName = "leetoya";
+        private const string Password = "0000";
+        private const string DoubleBackslash = "\\";
+        private const string Slash = "/";
+        private const string DoublePipe = "||";
+        private const string DotTxtString = ".txt";
+
+        #endregion
+
+        #region Property
+
+        #endregion
+
+        #endregion
+
+        #region Constructor
+
+        public UpdateClientController(FormUpdateClient formUpdateClient)
+        {
+            this.FormUpdateClient = formUpdateClient;
+
+        }
+
+        #endregion
+
+        #region Function
+
+        #region Private
+
+        private bool DoesFtpDirectoryExist(string existedfolder)
+        {
+            FtpWebRequest request = (FtpWebRequest)WebRequest.Create(RequestUriString);
+            request.Method = WebRequestMethods.Ftp.ListDirectory;
+            request.Credentials = new NetworkCredential(UserName, Password);
+
+            FtpWebResponse response = (FtpWebResponse)request.GetResponse();
+
+            Stream responseStream = response.GetResponseStream();
+
+            using (StreamReader reader = new StreamReader(responseStream))
+            {
+                string line;
+
+                while ((line = reader.ReadLine()) != null)
+                {
+                    if (line == existedfolder)
+                    {
+                        return true;
+                    }
```

Comment:
why don't you put 'break' once you found existing  folder. You don't need to wait till finish loop